src/Equations/ConfinementTime.cpp

```diff
@@ -55,11 +92,160 @@ real_t ConfinementTime::EvaluateParallelConfinementTime(len_t ir) {
 /**
  * Evaluates the perpendicular confinement time.
  */
-real_t ConfinementTime::EvaluatePerpendicularConfinementTime(len_t ir) {
-    real_t a      = radials->GetMinorRadius();
-    real_t B      = radials->GetMagneticField();
-    real_t T_cold = unknowns->GetUnknownData(id_Tcold)[ir];
-    return T_cold/(8*a*a*B);
+
+//Possible improvement using a lambda function
+//Return a power list containing the coeffcients for each law stored in a map in a .hpp file
+const PowerList ConfinementTime::GetCoeff(STREAM::OptionConstants::Conf_Time_type _type) const 
+{
+	return COEFFICIENTS::LawCoefficients.at(_type);
+}
+
+const PowerList ConfinementTime::GetCoeff() const //Return a power list containing the coeffcients for each law stored in a map in a .hpp file
+{
+	return COEFFICIENTS::LawCoefficients.at(type);
+}
+
+//Computation od the ohmic power
+real_t ConfinementTime::GetOhmicPower(len_t ir) const
+{
+	real_t V = eqsys->GetPlasmaVolume()->GetPlasmaVolume();
+	real_t E_field = unknowns->GetUnknownData(id_Efield)[ir];	
+	real_t sigma = eqsys->GetREFluid()->GetElectricConductivity(ir);
+	real_t R0 = radials->GetMajorRadius();
+	
+	return E_field * E_field * sigma * V / R0; // Need to normalize by R0 
+}
+
+//Goldstone power scaling law
+real_t ConfinementTime::Goldstone_scaling(PowerList const& coeff, len_t ir, real_t const& ConvUnit) const //
+{
+	real_t a = radials->GetMinorRadius();
+	real_t B0 = radials->GetMagneticField();
+	real_t R0 = radials->GetMajorRadius();
+	real_t kappa = radials->GetElongation();
+	
+	real_t Ip = ConversionFactor * unknowns->GetUnknownData(id_Ip)[ir]; // To convert into mega
+		
+	real_t ne = ConvUnit * unknowns->GetUnknownData(id_Ncold)[ir];  // to have the correct units 10^-19 or 10^-20 depending of the case
+	
+	real_t Pm = GetOhmicPower(ir) * ConversionFactor;
+	
+	real_t A = 2.0141017926; // ATOMIC MASS OF DEUTERIUM ----> Possible Improvement using IonHandler
+	
+	real_t prod = coeff[0]; // The first element is the constant
+	
+	const std::array<const real_t, 8> val = {B0, Ip, ne, a, R0, kappa, A, Pm};
+	for (size_t i = 0; i < val.size(); ++i)
+	{
+		prod *= pow(val[i], coeff[i+1]);
+	}
+	return prod;
+}
+
+// Bohm confinement time
+real_t ConfinementTime::Bohm_ConfinementTime(len_t ir) const 
+{
+	real_t a      = radials->GetMinorRadius();
+	real_t B      = radials->GetMagneticField();
+	real_t T_cold = unknowns->GetUnknownData(id_Tcold)[ir];
+	return T_cold / (8*a*a*B);
+}
+
+// INTOR scaling law
+real_t ConfinementTime::INTOR_ConfinementTime(len_t ir) const 
+{
+	real_t q = eqsys->GetSettings()->GetReal("timestep/safetyfactor");
```

Comment:
woudl just ask this once more, but please include a reference to each confinement law, or othwerise include a readme or link to the wiki for documentation. Thanks! sorry I know this is tedious...

Reply:
Done